go_lib/hooks/internal_tls/hook.go

```diff
@@ -121,17 +140,33 @@ func genSelfSignedTLS(conf GenSelfSignedTLSHookConf) func(input *go_hook.HookInp
 		} else {
 			// Certificate is in the snapshot => load it.
 			cert = input.Snapshots["secret"][0].(certificate.Certificate)
+			// update certificate if less than 6 month left. We create certificate for 10 years, so it looks acceptable
+			// and we don't need to create Crontab schedule
+			expiring, err := certificate.IsCertificateExpiringSoon([]byte(cert.Cert), 4380*time.Hour) // 6 month
```

Comment:
```suggestion
			expiring, err := certificate.IsCertificateExpiringSoon([]byte(cert.Cert), 4380*time.Hour) // 6 month
```
Let's make the IsCertificateExpiringSoon function accept a certificate as a string if it is more convenient.

Reply:
So, it was my mistake. This function is already used in several places.